diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 14b6dfaa73957a7832335206511e764cdd2ea730..0000000000000000000000000000000000000000
diff --git a/typings.d.ts b/lib/cjs/typings.d.ts
similarity index 100%
rename from typings.d.ts
rename to lib/cjs/typings.d.ts
diff --git a/lib/esm/typings.d.mts b/lib/esm/typings.d.mts
new file mode 100644
index 0000000000000000000000000000000000000000..580c56f4397ad8d7ba8a9527e95bbe3e973802b8
--- /dev/null
+++ b/lib/esm/typings.d.mts
@@ -0,0 +1,24 @@
+import type { ErrorObject } from 'ajv';
+
+export interface IOutputError {
+  start: { line: number; column: number; offset: number };
+  // Optional for required
+  end?: { line: number; column: number; offset: number };
+  error: string;
+  suggestion?: string;
+}
+
+export interface IInputOptions {
+  format?: 'cli' | 'js';
+  indent?: number | null;
+
+  /** Raw JSON used when highlighting error location */
+  json?: string | null;
+}
+
+export default function <S, T, Options extends IInputOptions>(
+  schema: S,
+  data: T,
+  errors: Array<ErrorObject>,
+  options?: Options
+): Options extends { format: 'js' } ? Array<IOutputError> : string;
diff --git a/package.json b/package.json
index baaa5940665ea50aebee3b93b822d79c62772b6b..7608634ec7c25e99586b4d39e5356eecc8bf7edc 100644
--- a/package.json
+++ b/package.json
@@ -6,8 +6,14 @@
   "main": "./lib/cjs/index.js",
   "exports": {
     ".": {
-      "require": "./lib/cjs/index.js",
-      "default": "./lib/esm/index.mjs"
+	"import": {
+          "types": "./lib/esm/typings.d.mts",
+          "default": "./lib/esm/index.mjs"
+      	},
+        "require": {
+          "types": "./lib/cjs/typings.d.ts",
+          "default": "./lib/cjs/index.js"
+        }
     }
   },
   "module": "./lib/esm/index.mjs",
@@ -26,10 +32,9 @@
     "Tong Li"
   ],
   "license": "Apache-2.0",
-  "types": "./typings.d.ts",
+  "types": "./lib/cjs/typings.d.ts",
   "files": [
-    "lib",
-    "typings.d.ts"
+    "lib"
   ],
   "scripts": {
     "prebuild": "rm -rf lib",
